### ISSUE TITLE
Dark extension for code examples (editors snaps where code is written)

### ISSUE THREAD

**Issue by shreyaswikriti** [ocaml/v2.ocaml.org#1431]
Programmers often like the dark theme. I found this eye-catching.
![Screenshot from 2021-04-06 13-35-28](https://user-images.githubusercontent.com/57635473/113681859-1ee38500-96e0-11eb-8a8f-861268f1560b.png)
rather than a light theme, we can convert the editor to a dark theme. I'll start working on this if you like. @patricoferris

**Comment by pitag-ha** [comment]
Having dark code snippets seems to break a bit with the rest of the site, which is light, doesn't it? Do you have an example of a site where that happens?

**Comment by shreyaswikriti** [comment]
![image](https://user-images.githubusercontent.com/57635473/113771636-4454ab00-9741-11eb-96f3-637e2b00c9f4.png)
it's the website of the iconic framework. They also use dark extension. 
Actually, I thought this extension look quite new and eye-catching that's why I suggested it. Thank you for reviewing this issue.

**Comment by pitag-ha** [comment]
In the screenshot you've sent everything is dark. The proposal of the issue is making only the code blocks dark and leaving the rest light, right?

**Comment by shreyaswikriti** [comment]
yes, I proposed to make only code blocks dark not all websites.
Just as the above screenshot I have sent (second one).

**Comment by pitag-ha** [comment]
Sorry, I've misinterpreted the (second) screenshot you've sent. I think this issue is a matter of taste :) Some sites have it one way, others have it the other way; some prefer it one way others prefer it the other way. 

Have you seen that applicants can only work on 2, max 3 simple issues (I'm referring to [this](https://github.com/ocaml/ocaml.org/issues/1245#issue-821019972))? I think you already have 2 simple issues (or more) to work on, don't you? Or do you rely on this one to have something simple? :)

**Comment by shreyaswikriti** [comment]
I have given only one simple issue to work on. I would love to contribute more.

**Comment by patricoferris** [comment]
Hi @shreyaswikriti,

I agree with @pitag-ha that this is definitely one of those subjective issues, but I think to add a dark colour format now would entail two important aspects:

 - The ability to toggle between the dark and light -- I can't think of a reason why switching to dark from light is an improvement, but having both probably is.
 - More work related to the accessibility of the colour choices -- I did quite a lot of work not too long ago in researching the colours we currently have (see https://github.com/ocaml/ocaml.org/issues/1199), to not regress on this I think we would want to do the same here. 

What do you think?

**Comment by pitag-ha** [comment]
> The ability to toggle between the dark and light?

Yes, I agree that that might be an improvement, if it is for the whole site. @patricoferris, do you think it would also make sense to only have dark mode for code examples? I think that's what the issue is about. @shreyaswikriti , please correct me if I got that wrong and you actually wanted to implement dark mode for the whole site :)

**Comment by shreyaswikriti** [comment]
Primarily, I raised this issue to change the color of code blocks only but as mentioned by @patricoferris theme switcher can be added to the whole site. I have seen many websites that do use the theme switcher.

**Comment by pitag-ha** [comment]
Sounds great. Could you send me a link to a couple of websites with theme switcher? It rings a bell having seen that somewhere, but I don't know where and I'm curious :)

**Comment by shreyaswikriti** [comment]
Hii @pitag-ha, there is a theme switcher on the GitHub website itself, where we can toggle between dark and light themes. Actually, theme switcher is famous on that website where users spend a lot of time like book reading websites, coding websites, etc. The reason being dark mode gives less eye strain. For this website, we have to Learn section where users can spend more time reading codes, learning about OCaml, etc. So it would be beneficial for a user to have a theme customize option. This is not a compulsory thing but an additional facility for users.

**Comment by shreyaswikriti** [comment]
So, Can I start working on this??

**Comment by pitag-ha** [comment]
> So, Can I start working on this??

Yes! :) Sounds good to me (and I'm pretty sure it also does to @patricoferris ). You're right: I should have made that clearer.

**Comment by shreyaswikriti** [comment]
_**I made a theme switcher for the site. Light mode is the same as previous.Since the video quality is not good, I am attaching some screenshots.**_
Kindly have a look @pitag-ha @patricoferris @gs0510

https://user-images.githubusercontent.com/57635473/114870998-712a5180-9e16-11eb-93fe-d05878ac224d.mp4
**Screenshots**
![Screenshot from 2021-04-15 18-12-08](https://user-images.githubusercontent.com/57635473/114871369-d8480600-9e16-11eb-8847-4bec3ca3c304.png)
![Screenshot from 2021-04-15 18-12-23](https://user-images.githubusercontent.com/57635473/114871415-e3029b00-9e16-11eb-8523-c8aea4361301.png)
![Screenshot from 2021-04-15 18-12-37](https://user-images.githubusercontent.com/57635473/114871470-eeee5d00-9e16-11eb-852f-a22a186faafd.png)
![Screenshot from 2021-04-15 18-12-40](https://user-images.githubusercontent.com/57635473/114871510-f877c500-9e16-11eb-87ba-ac1f4f26e2d2.png)
![Screenshot from 2021-04-15 18-12-52](https://user-images.githubusercontent.com/57635473/114871531-01689680-9e17-11eb-92cc-9e43cb6f0f30.png)
![Screenshot from 2021-04-15 18-13-34](https://user-images.githubusercontent.com/57635473/114871567-0a596800-9e17-11eb-86c1-90e58ee85b46.png)
![Screenshot from 2021-04-15 18-14-11](https://user-images.githubusercontent.com/57635473/114871607-13e2d000-9e17-11eb-8e7c-3d43ad35b922.png)
![Screenshot from 2021-04-15 18-13-42](https://user-images.githubusercontent.com/57635473/114871655-23621900-9e17-11eb-92c4-3aa406223af8.png)

**Comment by gs0510** [comment]
@shreyaswikriti This is a great start! I have a few comments:
1. The code block that's gray is harder to read than before, maybe you can choose a different gray that's help with legibility.
2. The toggle looks a little out of place with the emojis. It would look more consistent with a simpler toggle.
3. Can you run an accessibility tool with your website, I feel there's a bunch of places where the contrast is wonky and makes it harder to see.

**Comment by shreyaswikriti** [comment]
@gs0510 I changed the gray color.
![Screenshot from 2021-04-15 19-07-08](https://user-images.githubusercontent.com/57635473/114878397-d6ce0c00-9e1d-11eb-879f-5cc602901e49.png)
Also, I have some contrast errors. Can you suggest to me how do I fix those?

**Comment by shreyaswikriti** [comment]
@gs0510, **_I have made the changes as suggested.
Also, I used the 'WAVE Evaluation Tool ' and it's showing no contrast errors. Kindly have a look._**

https://user-images.githubusercontent.com/57635473/115051306-4d403c00-9efa-11eb-845e-4cb8f517b408.mp4













![Screenshot from 2021-04-16 21-21-03](https://user-images.githubusercontent.com/57635473/115051713-bfb11c00-9efa-11eb-8da1-4a85ef191fbc.png)
![Screenshot from 2021-04-16 21-21-09](https://user-images.githubusercontent.com/57635473/115051747-c93a8400-9efa-11eb-9e3f-11dd8471b148.png)
![Screenshot from 2021-04-16 21-21-22](https://user-images.githubusercontent.com/57635473/115051778-d192bf00-9efa-11eb-8876-cfdfd91360cc.png)
![Screenshot from 2021-04-16 21-22-15](https://user-images.githubusercontent.com/57635473/115051818-da839080-9efa-11eb-8fbe-3d72fb07ba86.png)
![Screenshot from 2021-04-16 21-21-32](https://user-images.githubusercontent.com/57635473/115051883-eb340680-9efa-11eb-9fc0-80ad79fe6815.png)

**Comment by gs0510** [comment]
@shreyaswikriti It definitely looks better! But I think every time you switch to another link, there's a flash where the bright version is loaded first and then the dark version is applied which makes browsing the website uncomfortable. Can you fix that?

The switch can  be something like this:
![Screenshot from 2021-04-16 18-11-19](https://user-images.githubusercontent.com/8718814/115053229-32ac9980-9edf-11eb-9e0e-50d1d7d1b585.png)

Thanks!

**Comment by shreyaswikriti** [comment]
@gs0510,

-  I searched about the first problem on google. I found this flash generates due to many problems. It's a complex issue that involves speeding up the website. By the way I am doing my best if I can find any solution. 
- I implemented the desired toggle button.

**Comment by shreyaswikriti** [comment]
@gs0510, Kindly  Have a look. thanks
https://user-images.githubusercontent.com/57635473/115125553-8c4bbb80-9fe6-11eb-9a78-d63df69a4e82.mp4

**Comment by gs0510** [comment]
Looks really good @shreyaswikriti!! you should go ahead and open a PR :)
